### PR TITLE
VGNCParser

### PR DESCRIPTION
**Pull request by tgrego**
## Description

Porting of the VGNC parser to the new repo.

## Testing

Tests are included

### CODE REVIEW

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  if ($verbose) {
+    print "Loaded a total of $count VGNC xrefs\n";
+  }
+
+  return 0; # successful
+}
+
+
+=head2 add_synonyms_for_hgnc
+  Description: Specialized class to add synonyms from HGNC and VGNC data
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: Specialized class?

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  Return type: N/A
+  Caller     : internal
+=cut
+
+sub add_synonyms_for_hgnc {
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: I see a similar method in HGNCParser (old repo) as well.   Wondering if there will be two copies of the same method in both HGNC and VGNC parsers?!

File: modules/t/vgnc.t [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  $db->schema->resultset('Xref')->check_direct_xref({
+    accession   => 'VGNC:14659',
+    label       => 'CYYR1',
+    description => 'cysteine and tyrosine rich 1',
+    source_id   => 144,
+    species_id  => 9598
+  }),
+ 'Sample chimpanzee direct Xref has been inserted'
+);
+
```
Comment (premanand17) [inline modules/t/vgnc.t]: Guess synonyms/alias creation can also be tested here?

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm (hunk 2 above)
Comment (magaliruffier) [reply modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: The VGNCParser is indeed a simplified copy of the HGNCParser, so this method can probably go into the BaseAdaptor

Comment (tgrego) [reply modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: No, I will either keep a single copy in HGNCParser (and change the parent of VGNCParser to there) or preferably move the method to the 'BaseParser' and call it through the $xref_dba

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  my $count = 0;
+
+  my $file_io = $xref_dba->get_filehandle($file);
+
+  if ( !defined $file_io ) {
+    confess "Can't open VGNC file $file\n";
+  }
+
+  # Create a hash of all valid taxon_ids for this species
+  my %species2tax = $xref_dba->species_id2taxonomy;
```
Comment (magaliruffier) [inline modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: the species_id argument provided to the run method is already the species taxonomy id, the extra lookup is not required
see https://github.com/Ensembl/ensembl-xref/pull/31

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm (hunk 1 above)
Comment (tgrego) [reply modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: moved to BaseAdaptor as add_synonyms_for_hgnc_vgnc()

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm (hunk 2 above)
Comment (tgrego) [reply modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: moved to BaseAdaptor as add_synonyms_for_hgnc_vgnc()

File: modules/t/vgnc.t (hunk 3 above)
Comment (tgrego) [reply modules/t/vgnc.t]: test now added

File: modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm (hunk 4 above)
Comment (tgrego) [reply modules/Bio/EnsEMBL/Xref/Parser/VGNCParser.pm]: removed taxonomy lookups

**Pull request merged**